### ISSUE TITLE
Split out implementation and interfaces into a separate packages

### ISSUE THREAD

**Issue by tugberkugurlu** [mxjones/RedBus#5]
It would be great to have a separate package for [Interfaces](https://github.com/mxjones/RedBus/tree/master/src/Redbus/Redbus/Interfaces) (e.g. `Redbus.Abstractions`). The main need I have for this is that I want my domain project to have a reference to `Redbus.Abstractions` package but don't want it to have a reference to concrete implementation of the event bus so that it prevents direct coupling to the implementation.

I am happy to contribute this but wanted to check with you first to see if this would be something you are up to having.

**Comment by mxjones** [comment]
@tugberkugurlu yes that sounds good to me 👍 - You can go ahead with the changes neccessary for this if you want and submit a PR and I will sort out the NuGet package related stuff, thanks!

**Comment by facundofarias** [comment]
True @tugberkugurlu! Same happening here!
